### ISSUE TITLE
Question on canonical representation for P4Data compound types

### ISSUE THREAD

**Issue by jafingerhut** [p4lang/p4runtime#154]
My apologies for not raising this question _before_ PR https://github.com/p4lang/p4runtime/pull/153 was merged in, but I didn't think of it until shortly after then.

This is what the updated code for message `P4Header` says:
```
message P4Header {
  // If the header is invalid (is_valid is "false"), then the repeated
  // bitstrings must be empty.
  bool is_valid = 1;
  repeated bytes bitstrings = 2;
}
```
Would it perhaps be preferable if instead of the bitstrings being empty, that the entire bitstrings field be unset?

The only reason I ask is that earlier when the encodings for signed and unsigned bitstrings is specified, it says: "If the string's byte length is zero, the server always rejects the string."  I know we could make an exception to that rule for encoding invalid headers, but was wondering whether it was a good idea to make such an exception.

**Comment by jafingerhut** [comment]
Or, perhaps by saying "the repeated bitstrings must be empty", you meant that there are 0 bitstrings in the repeated field, _not_ that there are N bitstrings, each of which is a length zero `byte` type value?

**Comment by antoninbas** [comment]
Yes it was supposed to say the "bitstrings repeated field" must be empty, like in the spec. I'll fix the phrasing in a future PR.

**Comment by antoninbas** [comment]
I included a fix for this in #156
